python/pandemic_simulator/environment/pandemic_env.py

```diff
@@ -46,6 +47,36 @@ def __init__(self,
         :param sim_steps_per_regulation: number of sim_steps to run for each regulation
         :param non_essential_business_location_ids: an ordered list of non-essential business location ids
         """
+        
+        
+        
+
+        self.action_space=spaces.Discrete(2) 
+        '''
+        action=0 --> Reduce level of regulation.
+        action=1 --> Maintain level of regulation.
+        action=2 --> Increase level of regulation.
+        '''
+
+        #Uncomment below line to make actions to choose regulation level
+        #self.action_space = gym.spaces.Discrete(len(self._stage_to_regulation))
+
+        self.observation_space=spaces.Tuple([
+            spaces.Discrete(len(pandemic_sim._persons)),         #Global Infection Summary None
```

Comment:
I'm not sure why we're doing this, but even then, can we just save the value in a variable and then use it? @varunrajk any ideas on why this is being added?

Reply:
observation_space is a variable implemented in gym environments. If we add this people can apply models they used for other environments to this as they use env.action_space and env.observation_space as dimensions for output and input layers for the neural network. Allowing a general code that can be used for all environments.

Reply:
Adding observation_space makes sense to get info for setting up models, buffer, etc. But I would suggest we switch to using spaces.Dict for this. For example, 
we could set this to something like 
```
self.observation_space = spaces.Dict({
            'global_infection_summary': spaces.Discrete(len(pandemic_sim._persons), 
            ...
        })
```
Also, can you add the following property to the simulator
```
@property
def num_persons(self) -> int:
      return len(self._persons)
```
Then in the pandemic_env you can just access via the public attribute `pandemic_sim.num_persons` instead of the current private `pandemic_sim._persons`.

Reply:
Hi sorry for the  delayed reply. I am currently having my final exams. I will get free by 12/22/21. I will make this change then, as I will have to change code for each time we access observation.

Reply:
I have done as requested. The current output still uses the old PandemicObservation Class I will resolve that. Would it be ok if I delete it?